tutorials/add_model.md

```diff
@@ -0,0 +1,139 @@
+```python
+from .recom_pmf import PMF
+```
+
+One last step is required to make you recommender model available to Cornac, which is to update `../models/__init__.py` by adding the following line:
+```python
+from .pmf import PMF
+```
+
+### 3. Indicating dependencies
+The file `requirements.txt` is optional as opposed to the others, and it is only needed if your implementation relies on some external libraries/frameworks, e.g., TensorFlow, PyTorch, etc., in which case you should include this file and indicate which versions of the framework(s) are required. Here is a sample of a `requirements.txt` file:
+```
+tensorflow>=1.10.0
+```
+
+### 4. Adding documentation
+
+Cornac uses [Sphinx](http://www.sphinx-doc.org/en/master/) to generate documentation. All you need to do is to fulfill the Parameters section inside `recom_pmf.py`. If you are not familiar with Sphinx syntax conventions, you can just take a look at some of the existing Cornac models and follow them.
+
+Once you have documented the parameters of your recommender, the next step, to integrate your model to the docs, is to update `cornac/docs/source/models.rst` by adding the following code:
+```
+Probabilitic Matrix Factorization (PMF)
+-----------------------------------------
+.. automodule:: cornac.models.pmf.recom_pmf
+   :members:
+```
+
+At this point you are done, and ready to create a pull request, congratulation!
+
+### Including an example (optional)
+We highly encourage you to add an example on how to fit your model to some dataset and report the obtained results, see for instance `./cornac/examples/pcrl_example.py`. All examples should be added to `./cornac/examples/`. 
+
+
+## Summary
+
+In short, contributing a recommender model to Cornac involves,
+
+- Creating new files
+    - [x] ./cornac/models/model_name/\_\_init__.py
+    - [x] ./cornac/models/model_name/recom_model_name.py
+    - [x] ./cornac/models/model_name/requirement.py (optional)
+- Redefining two functions
+     - [x] Recommender.fit()
+     - [x] Recommender.score()
+- Updating existing files
+     - [x] ./cornac/models/\_\_init__.py
+     - [x] ./docs/source/models.rst
+     
+As a concrete example, you can take a look at one of the Cornac's model implementations inside `./cornac/models/`, as this may help you save time.   
+    
+## Using Cython and C/C++ (optional)
+
+- If you are interested in using [Cython](https://cython.org/) to implement the algorithmic part of your model, then the Cornac's PMF implementation `./cornac/models/pmf` is a good example to look at.
+- If you already have a C/C++ implementation of your model, then you can use Cython to wrap your code for Python. In this case, you may consider `./cornac/models/hpf` or `./cornac/models/c2pf`, which are based on C++ implementations.
+
+Note that, in this situation you will need to add your Cython extension into `./setup.py`.
```

Comment:
Note that, in this situation, you will need to declare your Cython extension in the `./setup.py` file.